rir/src/ir/Compiler.cpp

```diff
@@ -122,13 +122,18 @@ class Context {
     void pushPromiseContext(SEXP ast) { code.push(new PromiseContext(ast, fun, code.empty() ? nullptr : code.top())); }
 
     Code* pop() {
-        auto res = cs().finalize(0);
+        Code* res = cs().finalize(0);
         delete code.top();
         code.pop();
         return res;
     }
 };
 
+// `true` if an argument isn't missing, labeled, or `...`.
+bool isRegularArg(RListIter& arg) {
```

Comment:
Since this is only used by `compileSpecialCall`, I would probably make it a lambda function, and define it inside `compileSpecialCall`, on line 145.

Something like:
```
auto isRegularArg = [](RListIter& arg) {
 return ...
};
```